### DISCUSSION TITLE
Asset reconciliation sensor use cases

### DISCUSSION THREAD

**Discussion by yeachan153** [dagster-io/dagster#10689] [Q&A]
Hi all,

Thanks to @sryza, I was able to try out the asset reconciliation sensor, specifically in our case for partitioned assets. There were a few questions that cropped up during testing w.r.t it's behaviour.

One of our initial use cases was that we would have hundreds of assets (potentially split off into multiple repos) that have different partitioning schemes and dependencies on each other. I guess there's two ways we can approach this:

- Use a sensor per asset, create our own partition mappings like https://docs.dagster.io/concepts/partitions-schedules-sensors/asset-sensors#monitoring-daily-assets-to-materialize-a-weekly-asset
- Use the asset reconciliation sensor to manage all assets

With the asset reconciliation sensor, passing hundreds of assets as arguments to the sensor creates quite a tight coupling and means that every asset would have to somehow be imported into one module and evaluated every-time someone added a new asset.

However, it does feel like the reconciliation sensor was created to remove some of the pain of creating multiple multi-asset sensors and mapping our own partitioning scheme. Is that correct? What were the main motivations/envisioned use cases? What would be the recommended approach to take in our case?

Thanks,
Yeachan

**Comment by sryza** [comment]
Hi @yeachan153 - in the long run, I'd definitely recommend the asset reconciliation sensor for this. We've started working on making it work across repositories, so not all the assets will need to be importable from the same module.  I'm hoping to land that in a month (although it might get a little stretched with the holidays). If it's important for you to have them in multiple repositories in the near term, then I'd recommend going with your own multi-asset sensor.
